### DISCUSSION TITLE
onClick" was passed to <Link> with `href` of `/` but "legacyBehavior" was set. The legacy behavior requires onClick be set on the child of next/link

### DISCUSSION THREAD

**Discussion by nishant-k1** [vercel/next.js#39212] [Help]
can someone please explain this? why would you get such a warning and how to get rid of it?

**Comment by JuanSystems** [comment]
amigo yo solucione el problema de esta forma:

```
<div className={`links ${clicked ? "active" : ""}`}>
            <Link href="/">
              <a onClick={handleClick}> Home</a>
            </Link>
            <Link href="/">
              <a onClick={handleClick}> Shop</a>
            </Link>
            <Link href="/">
              <a onClick={handleClick}> Blog</a>
            </Link>
          </div>
```
Espero que te ayude.

**Reply by nishant-k1** [reply]
Hi Juan, Thanks for your help. In my case, I don't see any use to define the handleClick function. It'd be more helpful if you could tell what is actually handled inside the handle click function here.

**Reply by jsfour** [reply]
One usecase would be click tracking. Eg you want to fire off an instrumentation event when the user clicks Home.

**Comment by SeinopSys** [comment]
I looked here on GitHub to see the commit that introduced this change and I was lead to #36436 which seems to have more information about this change. Could not find it in the docs.

**Comment by curiouscod3** [comment]
I have the same warning issue. 
I don't see any logical problem with my code.  Seems like this is a bug.
next ver. is  12.3.1

`<Menu.Item>
          {path ? (
         <Link href={path}>`

**Reply by n-glaz** [reply]
I am also having this issue specifically with HeadlessUI's `<Menu.Item>` component, it may expect to have the href set on in as there is a link in there, would have to look at the HeadlessUI docs

**Reply by iteahmad** [reply]
Menu item would attach an on click to its child component , to avoid this warning just wrap your link with a `span `

**Reply by hodgerpodger** [reply]
thanks!  @iteahmad suggestion worked for me.
